pkg/envoy/lds/response.go

```diff
@@ -44,6 +39,23 @@ func NewResponse(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_d
 
 	lb := newListenerBuilder(meshCatalog, svcAccount.ToServiceIdentity(), cfg, statsHeaders)
 
+	if pod, err := envoy.GetPodFromCertificate(proxy.GetCertificateCommonName(), meshCatalog.GetKubeController()); err != nil {
+		log.Warn().Msgf("Could not find pod for connecting proxy %s. No metadata was recorded.", proxy.GetCertificateSerialNumber())
+	} else if meshCatalog.GetKubeController().IsMetricsEnabled(pod) {
+		// Build Prometheus listener config
+		prometheusConnManager := getPrometheusConnectionManager()
+		if prometheusListener, err := buildPrometheusListener(prometheusConnManager); err != nil {
+			log.Error().Err(err).Msgf("Error building Prometheus listener for proxy %s", proxy.String())
+		} else {
+			ldsResources = append(ldsResources, prometheusListener)
+		}
+	}
+
+	if proxy.Kind() == envoy.KindGateway {
+		ldsResources = append(ldsResources, lb.newMultiClusterGatewayListener())
+		return ldsResources, nil
+	}
+
```

Comment:
Lets have this block where it was earlier and have the Gateway check after it

Reply:
makes sense. 

I don't see the multicluster feature flag being used anywhere, could we please wrap all code in the NewResponse for multicluster under its feature flag.

Reply:
done!